### PR TITLE
tests: Speedup feature_pruning test and refactor big transaction logic

### PR DESCRIPTION
**Pull request by conscott**
### Description 

Extended tests like feature_pruning.py create large blocks by making many large transactions with [a lot of OP_RETURNS](https://github.com/bitcoin/bitcoin/blob/master/test/functional/test_framework/util.py#L511-L527) in a single tx. Under this model roughly [14 of these transactions make a full-ish block](https://github.com/bitcoin/bitcoin/blob/master/test/functional/test_framework/util.py#L549-L560). 

These transactions are already non-standard, so I have just modified the logic to instead just create 1 large output, making a single large transaction, mining a block of roughly the same size (947k). The reduction in tx creation / number of outputs seems to significantly improve performance. 

Another optimization is to splice in the OP_RETURN output _after_ transaction signing, since it reduces the copying of the large OP_RETURN portion during signing. All signatures are `SIGHHASH_NONE`, so splicing after is okay. 

feature_pruning.py has about a 1.5x speedup on my machine (~12 minutes down to ~8 minutes), and `mempool_limit`, and `feature_maxuploadtarget` seem to benefit as well. 

### Refactoring
* The [original](https://github.com/bitcoin/bitcoin/blob/master/test/functional/test_framework/util.py#L511-L527) OP_RETURN generating function is now just a [string](https://github.com/conscott/bitcoin/blob/2018_11_opreturn_splices/test/functional/test_framework/util.py#L176-L192). If there is a better place to put this, I will move it. At the very least I just wanted to document the format because the previous function had little explanation for all the hex values. 
* `create_lots_of_big_transactions` no longer requires providing the OP_RETURN splice, and uses the default from before, with an optional `op_return_txout=` keyword args for using the giant OP_RETURN mentioned above.

### CODE REVIEW

**Comment by fanquake** [comment]
Concept ACK

Haven't looked at the changes, but I'm seeing a ~1.4x speedup on my machine between `feature_pruning.py` in master (e70a19e7132dac91b7948fcbfac086f86fec3d88) and c8ef522:
```
feature_pruning.py | ✓ Passed  | 819 s
ALL                | ✓ Passed  | 819 s (accumulated) 
Runtime: 819 s
```

```
feature_pruning.py | ✓ Passed  | 577 s
ALL                | ✓ Passed  | 577 s (accumulated) 
Runtime: 577 s
```

**Comment by fanquake** [comment]
@conscott I realise this hasn't seen much traction / concept ACKs, however are you interested in rebasing?

**Comment by conscott** [comment]
I almost forgot this was still alive :)

I will rebase when I get the chance, hopefully this week!

**Comment by sipa** [comment]
Concept ACK

File: test/functional/test_framework/util.py [hunk 1]
```diff
@@ -506,57 +553,40 @@ def create_confirmed_utxos(fee, node, count):
...
+    # 8 characters forwards past 'ffffffff' and the splice includes the correct number of outputs
+    # so the '01' can be skipped when copying the rest of the signed tx
+    signresult = signresult[:output_index+8] + op_return_txout + signresult[output_index+10:]
+    return node.sendrawtransaction(signresult, True)
+
+
+# Create a spend of each passed-in utxo, splicing in "txout" to each raw
+# transaction to make it large. By default, it will splice in
+# 128 OP_RETURN outputs of 512 data bytes (see MULTI_OP_RETURN)
+def create_lots_of_big_transactions(node, utxos, num, fee, op_return_txout=MULTI_OP_RETURN):
```
Comment (promag) [inline test/functional/test_framework/util.py]: Could simplify by ditching `num` parameter? This way below could be
```py
return [create_big_transaction(node, utxo, fee, op_return_txout) for utxo in utxos]
```

**Review by promag — COMMENTED** [review]
Concept ACK.

**Comment by jamesob** [comment]
Concept ACK. Will review on rebase.

**Comment by jnewbery** [comment]
Is this still required after #15686?

**Comment by conscott** [comment]
@jnewbery - Looks like you got it covered more elegantly with OP_NOPs :) It was a fun learning experience anyhow. 

I will go ahead and close this.

**Pull request closed without merging**